src/compute/src/bin/computed.rs

```diff
@@ -119,9 +121,10 @@ async fn run(args: Args) -> Result<(), anyhow::Error> {
             "serving internal HTTP server on {}",
             args.internal_http_listen_addr
         );
-        mz_ore::task::spawn(
-            || "computed_internal_http_server",
-            axum::Server::bind(&args.internal_http_listen_addr).serve(
+        mz_ore::task::spawn(|| "computed_internal_http_server", {
+            let listener = TcpListener::bind(args.internal_http_listen_addr)?;
+            socket::setsockopt(listener.as_raw_fd(), ReusePort, &true)?;
```

Comment:
@benesch the last non-bug panic that I saw was due to "address already in use" error when a storaged starts up using a particular HTTP port to listen to that was just vacated from some other storaged and was thus still in TCP_WAIT state. Hence the need to set `ReusePort` to make the panic go away.

Reply:
So `SO_REUSEADDR` _is_ set for the socket created by `TcpListener`:

https://github.com/rust-lang/rust/blob/1b225414f325593f974c6b41e671a0a0dc5d7d5e/library/std/src/sys_common/net.rs#L395-L403

I'm a little spooked by this. We never fixed the bugs in the process orchestrator that cause occasional port collisions, and I think `SO_REUSEPORT` is letting us create two `computed` processes that are binding to the same internal HTTP port simultaneously...

Reply:
@benesch  You are right , SO_REUSEADDR  indeed should have been set with the existing code already. However, without the patch, I can not get the Nightly parallel-* jobs to run panic-free. With my patch, they do. No idea why :-(

I briefly looked at the process orchestrator's port generation algorithm and it seems to be properly serialized behind a lock. There are two solutions that I can think of that I can try to implement myself:
- catch the "address already in use" error in a way that does not cause a panic but instead an exit with a non-crashing exit code 
- remove all port reuse from the process orchestrator and have it always assign monotonically increasing port numbers without ever going back (currently, the implementation seems to have facilities to release port numbers back to the "pool" (but it is not really a pool, as it does not seem to maintain a set of available ports, just the last generated port number) . Then we give the process orchestrator a very large maximum port value and hope for the best.

Please advise.